packages/calypso-e2e/src/lib/components/editor-toolbar-component.ts

```diff
@@ -0,0 +1,228 @@
+
+export type PreviewOptions = 'Desktop' | 'Mobile' | 'Tablet';
+
+const panel = 'div.interface-interface-skeleton__header';
+const selectors = {
+	// Block Inserter
+	blockInserterButton: `${ panel } button.edit-post-header-toolbar__inserter-toggle`,
+
+	// Draft
+	saveDraftButton: `${ panel } button.editor-post-save-draft[aria-disabled="false"]`,
+	saveDraftDisabledButton: `${ panel } button.editor-post-saved-state.is-saved[aria-disabled="true"]`,
+	switchToDraftButton: `${ panel } button.editor-post-switch-to-draft`,
+
+	// Preview
+	mobilePreviewButton: `${ panel } a.editor-post-preview`,
+	desktopPreviewButton: `${ panel } button.block-editor-post-preview__button-toggle`,
+	desktopPreviewMenuItem: ( target: PreviewOptions ) =>
+		`button[role="menuitem"] span:text("${ target }")`,
+	previewPane: ( target: PreviewOptions ) => `.is-${ target.toLowerCase() }-preview`,
+
+	// Publish
+	publishButton: `${ panel } button.editor-post-publish-button__button[aria-disabled="false"]`,
+
+	// Editor settings
+	settingsButton: `${ panel } .interface-pinned-items > button:first-child`,
+};
+
+/**
+ * Represents an instance of the WordPress.com Editor's navigation sidebar.
+ * The component is available only in the Desktop viewport.
+ */
+export class EditorToolbarComponent {
+	private page: Page;
+	private frameLocator: FrameLocator;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 * @param {FrameLocator} frameLocator Locator of the editor iframe.
+	 */
+	constructor( page: Page, frameLocator: FrameLocator ) {
+		this.page = page;
+		this.frameLocator = frameLocator;
+	}
+
+	/* General helper */
+
+	/**
+	 * Given a selector, determines whether the target button/toggle is
+	 * in an expanded state.
+	 *
+	 * If the toggle is in the on state or otherwise in an expanded
+	 * state, this method will return true. Otherwise, false.
+	 *
+	 * @param {string} selector Target selector.
+	 * @returns {Promise<boolean>} True if target is in an expanded state. False otherwise.
+	 */
+	async targetIsOpen( selector: string ): Promise< boolean > {
+		const locator = this.frameLocator.locator( selector );
+		const pressed = await locator.getAttribute( 'aria-pressed' );
+		const expanded = await locator.getAttribute( 'aria-expanded' );
+		return pressed === 'true' || expanded === 'true';
+	}
+
+	/* Block Inserter */
+
+	/**
+	 * Opens the block inserter.
+	 */
+	async openBlockInserter(): Promise< void > {
+		if ( ! ( await this.targetIsOpen( selectors.blockInserterButton ) ) ) {
+			const locator = this.frameLocator.locator( selectors.blockInserterButton );
+			await locator.click();
+		}
+	}
+
+	/**
+	 * Closes the block inserter.
+	 */
+	async closeBlockInserter(): Promise< void > {
+		if ( await this.targetIsOpen( selectors.blockInserterButton ) ) {
+			const locator = this.frameLocator.locator( selectors.blockInserterButton );
+			await locator.click();
+		}
+	}
```

Comment:
This is really nicely designed throughout this class 🙌  -- I love these juxtaposed methods that guarantee their ending state!! 🎉